rust/worker/src/server.rs

```diff
@@ -137,454 +135,230 @@ impl WorkerServer {
-                let proto_result = chroma_proto::VectorQueryResult {
-                    id: query_result.record.id,
-                    distance: query_result
-                        .distance
-                        .expect("The distance should be present"),
-                    vector: match query_result.record.embedding {
-                        Some(vector) => {
-                            match (vector, ScalarEncoding::FLOAT32, embedding_dim).try_into() {
-                                Ok(proto_vector) => Some(proto_vector),
-                                Err(e) => {
-                                    return Err(Status::internal(format!(
-                                        "Error converting vector: {}",
-                                        e
-                                    )));
-                                }
-                            }
-                        }
-                        None => None,
-                    },
-                };
-                proto_results.push(proto_result);
-            }
-            proto_results_for_all.push(chroma_proto::VectorQueryResults {
-                results: proto_results,
-            });
-        }
-
-        let resp = chroma_proto::QueryVectorsResponse {
-            results: proto_results_for_all,
-        };
-
-        Ok(Response::new(resp))
+        ))
     }
 
-    async fn get_vectors_instrumented(
+    async fn orchestrate_count(
         &self,
-        request: Request<GetVectorsRequest>,
-    ) -> Result<Response<GetVectorsResponse>, Status> {
-        let request = request.into_inner();
-        let segment_uuid = to_segment_uuid(&request.segment_id)?;
-        let collection_uuid = to_collection_uuid(&request.collection_id)?;
-        let (collection_version, log_position) = get_version_context(&request.version_context)?;
-
-        let orchestrator = GetVectorsOrchestrator::new(
+        count: Request<CountPlan>,
+    ) -> Result<Response<CountResult>, Status> {
+        let scan = count
+            .into_inner()
+            .scan
+            .ok_or(Status::invalid_argument("Invalid Scan Operator"))?;
+        let (fetch_log_operator, fetch_segment_operator) = self.decompose_proto_scan(scan)?;
+
+        // TODO: Update Count Orchestractor
+        let counter = CountQueryOrchestrator::new(
             self.clone_system()?,
-            request.ids,
-            segment_uuid,
-            collection_uuid,
-            self.log.clone(),
-            self.sysdb.clone(),
+            &fetch_segment_operator.metadata_uuid.0,
+            &fetch_log_operator.collection_uuid,
+            fetch_log_operator.log_client,
+            fetch_segment_operator.sysdb,
             self.clone_dispatcher()?,
             self.blockfile_provider.clone(),
-            collection_version,
-            log_position,
+            fetch_segment_operator.collection_version,
+            fetch_log_operator.start_log_offset_id as u64 - 1,
```

Comment:
discussed offline - lets not do this halfway

Reply:
Update: now it directly derive the fields from the given plan